### PR TITLE
Date: Mark getSettings as experimental

### LINKED ISSUE #10627: Date: Mark getSettings as experimental
As discussed in [7669](https://github.com/WordPress/gutenberg/pull/7996/files#r224152849) `getSettings()` in `wp-date` should be marked as experimental

### PR DESCRIPTION
**Pull request by ajitbohra**
Fixes #10627

### CODE REVIEW

**Review by Soean — CHANGES_REQUESTED** [review]
If we deprecate `getSettings()`, we should change all parts where we import and use it.

File: packages/date/src/index.js [hunk 1]
```diff
@@ -85,7 +90,17 @@ export function setSettings( dateSettings ) {
...
  */
+export function __experimentalGetSettings() {
+	return settings;
+}
+
+// deprecations
 export function getSettings() {
+	deprecated( 'wp.date.getSettings', {
+		version: '4.2',
+		alternative: 'wp.date.__experimentalGetSettings',
```
Comment (aduth) [inline packages/date/src/index.js]: Given the verbiage logged with the deprecation notice ("Please use ... instead"), we should consider either omitting the alternative or adding a [scary hint](https://github.com/WordPress/gutenberg/pull/9106/files#diff-e373e566d15f8f4196eb0b3998adbdd7R119) to discourage its use.

**Comment by ajitbohra** [comment]
@Soean ahh totally missed that deprecating for the first time :sweat_smile:

@aduth added hint to discourage the usage. If i am not wrong we need to update the changelog and bump the version, just noticed changlelog last version is `2.1.0` and package.json version is `2.0.2` !

**Review by aduth — COMMENTED** [review]
Code-wise this looks good, but it missed the 4.1.0 release so we'll need to bump back the deprecated version.

**Comment by ajitbohra** [comment]
@aduth 

version bumped 

need help on this 

> changelog the last version is 2.1.0 and package.json version is 2.0.2!

**Comment by aduth** [comment]
@ajitbohra Ah, sorry for missing the question. We never need to touch the `package.json` in any pull request, and it's expected the version would lag behind if there are "Unreleased" changes. We should include the note in `CHANGELOG.md` for the deprecation though, as a minor release.

See: https://github.com/WordPress/gutenberg/blob/master/CONTRIBUTING.md#maintaining-changelogs

**Comment by ajitbohra** [comment]
Note added to change log

File: docs/reference/deprecated.md [hunk 2]
```diff
@@ -23,6 +26,7 @@ Gutenberg's deprecation policy is intended to support backwards-compatibility fo
 - `wp.components.CodeEditor` has been removed. Used `wp.codeEditor` directly instead.
 - `wp.blocks.setUnknownTypeHandlerName` has been removed. Please use `setFreeformContentHandlerName` and `setUnregisteredTypeHandlerName` instead.
 - `wp.blocks.getUnknownTypeHandlerName` has been removed. Please use `getFreeformContentHandlerName` and `getUnregisteredTypeHandlerName` instead.
+- `wp.date.getSettings` has been removed. Please use `wp.date.__experimentalGetSettings` instead.
```
Comment (Soean) [inline docs/reference/deprecated.md]: You can remove this change from 4.2

Comment (aduth) [inline docs/reference/deprecated.md]:
```suggestion
```

Comment (ajitbohra) [reply docs/reference/deprecated.md]: oops

**Review by Soean — APPROVED** [review]
LGTM

File: packages/date/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+## 2.2.0 (2018-10-24)
```
Comment (Soean) [inline packages/date/CHANGELOG.md]:
```suggestion
## 2.2.0 (unreleased)
```

Comment (Soean) [reply packages/date/CHANGELOG.md]: The date is added on release, see https://github.com/WordPress/gutenberg/blob/master/CONTRIBUTING.md#maintaining-changelogs

Comment (ajitbohra) [reply packages/date/CHANGELOG.md]: Thank you, with this PR package updates and deprecation are lot clear to me :)

**Review by aduth — APPROVED** [review]
In some final testing I came across fact that we export `moment` as well, which (for different reasons than this) is probably something we don't want to expose on the API. Thoughts?

Not blocking for this pull request.

**Comment by youknowriad** [comment]
> In some final testing I came across fact that we export moment as well, which (for different reasons than this) is probably something we don't want to expose on the API. Thoughts?

We should definitely set "moment" as experimental (or get rid of it) as well. This is something I wanted to do but not sure if I'll have the time to do it.

**Comment by ajitbohra** [comment]
Yes, I remember the previous discussion around not exposing moment. Will have a look and work on the PR for the same.

**Pull request merged**